sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -313,10 +315,19 @@ abstract class RapidsShuffleInternalManagerBase(conf: SparkConf, val isDriver: B
       dependency: ShuffleDependency[K, V, C]): ShuffleHandle = {
     // Always register with the wrapped handler so we can write to it ourselves if needed
     val orig = wrapped.registerShuffle(shuffleId, dependency)
-    if (!shouldFallThroughOnEverything && dependency.isInstanceOf[GpuShuffleDependency[K, V, C]]) {
-      val handle = new GpuShuffleHandle(orig,
+    var shouldUseRapidsShuffle = false
+
+    if (!shouldFallThroughOnEverything) {
+      shouldUseRapidsShuffle = dependency match {
+        case gpuDependency: GpuShuffleDependency[K, V, C] =>
+          gpuDependency.useRapidsShuffle
+        case _ => false
+      }
+    }
+
+    if (shouldUseRapidsShuffle) {
```

Comment:
It's tempting to ditch the boolean entirely and just match directly to the dependency, e.g.:
```scala
  dependency match {
    case _ if shouldFallThroughOnEverything => orig
    case d: GpuShuffleDependency[K, V, C] if d.useRapidsShuffle =>
      new GpuShuffleHandle(orig, d.asInstanceOf[GpuShuffleDependency[K, V, V]])
    case _ => orig
  }
```

Reply:
ok went this route.